### DISCUSSION TITLE
Authentication using docker

### DISCUSSION THREAD

**Discussion by gr-qft** [supabase/supabase#4421] [Questions]
Hi! 

I'm starting my development using supabase docker. I don't see the same UI on my http://localhost:3000/ as on supabase.io, in particular the place to select select different auth providers. Is it possible to enable authentication (magic link, github etc.) with supabase docker and could you please point me to somewhere where I can read up a bit? 

Thanks!

**Comment by joeczucha** [comment]
According to https://github.com/supabase/supabase/discussions/4444, it sounds like the stuff under settings isn't included in the self-hosted version yet and that, when it is, it will be done through config files rather than through the dashboard.

**Comment by MrYakobo** [comment]
@gr-qft

You can actually access the settings if you navigate manually to http://localhost:3000/project/default/auth/settings :+1: 

![supabase](https://user-images.githubusercontent.com/11942597/152516070-2f3900f6-84bb-4aea-b489-1aeb5fa39238.png)

The link to settings is hidden, that might be because they are not fully operational right now. When I try to enter some details into the page and then click save, I get a 405 Method Not Allowed. PATCH method is not allowed.

![supabase-2](https://user-images.githubusercontent.com/11942597/152516509-518af844-cba1-4241-91b4-9d3d3d57d63e.png)

![supabase-3](https://user-images.githubusercontent.com/11942597/152516506-5314d1fb-f793-4c3d-91f8-e5782c5b4d13.png)

The request goes to the next.js server at `:3000`, so there might be some issue there. I can also reproduce it with `curl -X PATCH http://localhost:3000/api/auth/default/config`

Right now, for me, this is what's hindering me from switching to Supabase 100% :smile:

**Reply by MrYakobo** [reply]
Ok, so after some more digging, I found the offending `config.ts` next.js file.

https://github.com/supabase/supabase/blob/df382efa7652f47eecb71b136b16ed4185d0999b/studio/pages/api/auth/%5Bref%5D/config.ts

It's simply placeholders for now. `handlePost` does nothing and has the comment `// Platform only API`, so I guess we'll wait and see when that config file option comes. So you can simply ignore my replies here, lol
